helix-term/src/config.rs

```diff
@@ -19,16 +19,21 @@ pub struct LspConfig {
     pub display_messages: bool,
 }
 
-#[derive(Debug, Clone, PartialEq, Deserialize)]
+#[derive(Debug, Default, Clone, PartialEq, Deserialize)]
 #[serde(rename_all = "kebab-case")]
 pub struct TerminalConfig {
+    #[serde(default = "mouse_default")]
     pub mouse: bool,
+    #[serde(default = "middle_click_paste_default")]
+    pub middle_click_paste: bool,
+}
+
+fn mouse_default() -> bool {
+    true
 }
 
-impl Default for TerminalConfig {
-    fn default() -> Self {
-        Self { mouse: true }
-    }
+fn middle_click_paste_default() -> bool {
+    true
 }
```

Comment:
I think we should just create a custom impl, we shouldn't create a new function for every item.

```rust
impl Default for TerminalConfig {
    fn default() -> TerminalConfig {
        TerminalConfig {
            mouse: true,
            middle_click_paste: true,
        }
    }
}

Reply:
That way I got errors parsing config file with only one of those props (another is undefined):
```toml
[terminal]
middle-click-paste = false
```

Reply:
This was the way I did initially, but that way we set default for `terminal` property, but once another is provided it's validated, then fails because struct is not complete

Reply:
You can derive serde's `default` on the struct itself and then implement `Default` for the whole struct like pickfire suggested:

```rust
#[serde(rename_all = "kebab-case", default)]
pub struct TerminalConfig {
    pub mouse: bool,
    pub middle_click_paste: bool,
}
```

Reply:
Yes, works nice, thanks! Did that in https://github.com/helix-editor/helix/pull/554/commits/64daa604bfecf30f308b31c484c3c6fbedee226b